app/src/main/java/com/vero/tasky/agenda/data/repository/AgendaRepositoryImpl.kt

```diff
@@ -40,49 +40,45 @@ class AgendaRepositoryImpl(
     }
 
     override suspend fun syncAgenda(
-        deletedEventsIds: List<String>,
-        deletedTasksIds: List<String>,
-        deletedReminderIds: List<String>
-    ): Result<Unit> = ApiCallHandler.safeApiCall {
+        deletedEventIds: Array<String>,
+        deletedTaskIds: Array<String>,
+        deletedReminderIds: Array<String>
+    ): Result<Unit> = safeApiCall {
         api.syncAgenda(
             SyncAgendaRequest(
-                deletedEventsIds = deletedEventsIds,
-                deletedTasksIds = deletedTasksIds,
+                deletedEventIds = deletedEventIds,
+                deletedTaskIds = deletedTaskIds,
                 deletedReminderIds = deletedReminderIds
             )
         )
     }
 
-    override suspend fun getFullAgenda() = ApiCallHandler.safeApiCall {
+    override suspend fun getFullAgenda() = safeApiCall {
         val result = api.getFullAgenda()
         saveAgendaItems(result)
     }
 
     private fun getAgendaItemsFromDao(timestamp: Long) : List<AgendaItem> {
         val agendaItems = mutableListOf<AgendaItem>()
```

Comment:
Personally I don't tend to use mutableLists that much

I like to just save the items into variables

Again, this is 100% optional. I'm just letting you know how I prefer doing them

I would replace this method with:

```
val tasks = taskDao.loadTasksForDay(timestamp).map { it.toTask() }
val events = eventDao.loadEventsForDay(timestamp).map {it.toEvent() }
val reminders = reminderDao.loadRemindersForDay(timestamp).map { it.toReminder() }

return tasks + events + reminders 
```